blustl/encode.py

```diff
@@ -0,0 +1,82 @@
+from functools import singledispatch
+import operator as op
+
+import numpy as np
+from lenses import lens
+
+import stl
+import pyeda
+
+@singledispatch
+def encode_stl(stl):
```

Comment:
If we assume that the STL has been discretized (and removed temporal operators), then if we just change the atomic predicates to unique variables (maybe use a lens), then we can use the same to string trick that you use in `decode`.

Something like:

```python
def encode(phi:"STL"):
    # Encode STL as MTL and record map from AP to LinEq
    # This requires creating a unique id for each LinEq and replacing with stl.ast.AtomicPred
    # Should be easy with lens
    psi, AP_map = to_mtl(phi)

    # use correct logical AND and OR symbols
    psi_str = str(phi).replace("∨", "|").replace("∧", "&").replace("¬", "~")

    # parse using pyeda
    return expr(psi_str)
```